conans/client/loader.py

```diff
@@ -437,13 +438,20 @@ def _parse_conanfile(conan_file_path):
         with chdir(current_dir):
             old_dont_write_bytecode = sys.dont_write_bytecode
             sys.dont_write_bytecode = True
-            loaded = imp.load_source(module_id, conan_file_path)
+            try:
+                # FIXME: imp is deprecated in favour of implib
+                loaded = imp.load_source(module_id, conan_file_path)
```

Comment:
I think I would leave the ``sys.dont_write_bytecode`` around the ``load_source()``, otherwise it might be left in inconsistent state if there is a validate version exception raised.